examples/chip-tool/commands/common/Commands.cpp

```diff
@@ -74,8 +79,29 @@ int Commands::Run(int argc, char ** argv)
     err = mController.SetUdpListenPort(mStorage.GetListenPort());
     VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(Controller, "Init failure! Commissioner: %s", chip::ErrorStr(err)));
 
-    err = mController.Init(localId, initParams);
-    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(Controller, "Init failure! Commissioner: %s", chip::ErrorStr(err)));
+    VerifyOrExit(rcac.Alloc(chip::Controller::kMaxCHIPDERCertLength), err = CHIP_ERROR_NO_MEMORY);
+    VerifyOrExit(noc.Alloc(chip::Controller::kMaxCHIPDERCertLength), err = CHIP_ERROR_NO_MEMORY);
+    VerifyOrExit(icac.Alloc(chip::Controller::kMaxCHIPDERCertLength), err = CHIP_ERROR_NO_MEMORY);
+
+    {
+        chip::MutableByteSpan nocSpan(noc.Get(), chip::Controller::kMaxCHIPDERCertLength);
+        chip::MutableByteSpan icacSpan(icac.Get(), chip::Controller::kMaxCHIPDERCertLength);
+        chip::MutableByteSpan rcacSpan(rcac.Get(), chip::Controller::kMaxCHIPDERCertLength);
+
+        chip::Crypto::P256Keypair ephemeralKey;
+        SuccessOrExit(err = ephemeralKey.Initialize());
+
+        err = mOpCredsIssuer.GenerateNOCChainAfterValidation(localId, 0, ephemeralKey.Pubkey(), rcacSpan, icacSpan, nocSpan);
```

Comment:
Shouldn't this only happen when we are doing a new pairing, or better yet when we don't have stored credentials already?  Doing it here means generating new opcreds on every single run, but we only know the desired fabric id in the pairing commands...

This is probably OK for now as an incremental step if you don't want to detangle these bits, but in that case please make sure that there's a followup to get this right.

Reply:
Yes, right now it's ok since fabric ID is being set to 0. We can store the credentials on first run, and reuse them afterwards. Do we have a command parameter to chip-tool to set the fabric ID? I can try to hook it up. If not, that can go as a follow on PR.

Reply:
We have such a parameter, but only for the pairing commands.  Followup PR is just fine; let's not scope-creep this.

Reply:
I've added a todo for this item. The next PR will update it store the credentials in storage, and use the fabric ID.